tools/github/setup_test_env.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+# Options:
+#
+# MINIMUM_REQUIREMENTS:
+#   Install the minimum versions of all requirements, instead of the latest.
+#
+# OPTIONAL_DEPS:
+#   Install optional requirements.
+#
+# WITHOUT_POOCH:
+#   Remove pooch from environment.
+#
+# PIP_FLAGS:
+#   These options are passed to pip.
+
+# TODO: Remove special handling of free-threaded dependencies below.
+
+set -ex
+
+export PIP_DEFAULT_TIMEOUT=60
+python -m pip install --upgrade pip
+
+if [[ $MINIMUM_REQUIREMENTS == 1 ]]; then
+    for filename in requirements/*.txt; do
+        sed -i 's/>=/==/g' "$filename"
+    done
+fi
+
+# Combine requirement files for a more robust pip solve
+# installing successively may update previously constrained dependencies
+REQUIREMENT_FILES="-r requirements/default.txt -r requirements/test.txt"
+if [[ "${OPTIONAL_DEPS}" == "1" ]]; then
+    REQUIREMENT_FILES="${REQUIREMENT_FILES} -r requirements/optional.txt"
+fi
+
+python -m pip install $PIP_FLAGS $REQUIREMENT_FILES
+
+# TODO: delete when scipy, numpy, cython and pywavelets free-threaded wheels are available on PyPi
+FREE_THREADED_BUILD="$(python -c"import sysconfig; print(bool(sysconfig.get_config_var('Py_GIL_DISABLED')))")"
+if [[ $FREE_THREADED_BUILD == "True" ]]; then
+    pip install --pre -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple cython numpy scipy pywavelets
```

Comment:
Do we need Cython during testing?

```suggestion
    pip install --pre -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple numpy scipy pywavelets
```

Reply:
Still curious about this. But not critical I think since we plan to remove this eventually anyway.

Reply:
Yes, we should get rid of this as soon as free-threaded builds are more commonly available.